### DISCUSSION TITLE
Dashboard metrics stored in prometheus.

### DISCUSSION THREAD

**Discussion by snoby** [cybertec-postgresql/pgwatch2#805] [Q&A]
I have followed the guides and i have pgwatch2 scraping and sending metrics to prometheus.  Now I'm confused as to how / what image to use to get the grafana dashboards to function.  I know that i can't just pull in the grafana dashboards checked into source control because they don't ask for the data source.  I'm sure it's something simple that i'm missing but can't seem to find anything in the docs.
thanks,

**Comment by pashagolub** [comment]
Hello.

Have you checked this [chapter](https://pgwatch2.readthedocs.io/en/latest/custom_installation.html#custom-install-grafana)? Precisely item number 5.

**Reply by snoby** [reply]
Trying to work this out in my head...


`This could be done by importing the pgwatch2 dashboard definition JSON-s manually, one by one, from the “grafana_dashboards” folder (“Import Dashboard” from the Grafana top menu) or via as small helper script located at /etc/pgwatch2/grafana-dashboards/import_all.sh. The script needs some adjustment for metrics storage type, connect data and file paths.`

This is telling me I need to startup  any  postgres database put the information at the top of  the import_all.sh script file and then  change the GRAFANA_DB_TYPE to prometheus.

Then  connect grafana to that postgres and read the dashboards out?  
Or is it i'm just using that postgres database regardless  of how the data is collected?

Thanks,

**Reply by pashagolub** [reply]
I was typing enhanced answer but deleted it accidentally. 🤦 :)
 I will just retype the workaround part. 

> [!NOTE]
> The script is outdated and doesn't work well for modern Grafana versions.

Please, try to copy [grafana_dashboards/prometheus/v8](https://github.com/cybertec-postgresql/pgwatch2/tree/master/grafana_dashboards/prometheus/v8) content folder to the [provisioning/dashboards](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#dashboards) Grafana folder. It will catch them all up.
